uniswap-v2-core/contracts/UniswapV2Pair.sol

```diff
@@ -2,19 +2,31 @@ pragma solidity =0.5.16;
 
 import './interfaces/IUniswapV2Pair.sol';
 import './UniswapV2ERC20.sol';
-import './libraries/Math.sol';
-import './libraries/UQ112x112.sol';
-import './interfaces/IERC20.sol';
 import './interfaces/IUniswapV2Factory.sol';
 import './interfaces/IUniswapV2Callee.sol';
+import './libraries/Math.sol';
+import './interfaces/IERC20.sol';
+import './libraries/UQ112x112.sol';
+
 
 contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
     using SafeMath  for uint;
     using UQ112x112 for uint224;
 
     uint public constant MINIMUM_LIQUIDITY = 10**3;
     bytes4 private constant SELECTOR = bytes4(keccak256(bytes('transfer(address,uint256)')));
+    uint256 public constant SQUARED_ACCURACY = 100_000_000_000_000_000_000_000_000_000_000_000_000; // 100000000000000000000000000000000000000
```

Comment:
This value is currently the power of 10 right before breaching 2^128. We could look at slightly breaching 2^128, such that we guarantee no precision loss on division of two numbers if both are < 2^128. This depends on what the size of the balances we're dividing are. If they're less than the current `SQUARED_ACCURACY` - no change will be required.

Reply:
```
    function _mintFee(uint112 _reserve0, uint112 _reserve1) private returns (bool feeOn) {
            uint _sqrtNewK = Math.sqrt(uint(_reserve0).mul(_reserve1));
            uint _sqrtOldK = Math.sqrt(kLast); // gas savings
    }
```

So the inputs will never be above uint112 (assumng sqrt(uint112 * uint112) < uint112). `SQUARED_ACCURACY` should likely be increased to just below `uint256(-1)` so that we can get 256 - 112 bits of accuracy in the division result? `ACCURACY` would then be just below uint128 but still safe because it is only ever multiplied with a uin112

Reply:
With the accuracy for the calc now increased and proven with base-line test cases, we can likely close out this review point: though further platformFee testing will help confirm it further.

Reply:
Agreed, `SQUARED_ACCURACY` is now at the maximum value before overflowing and no obvious improvements come to mind.